### PR TITLE
Feature/323 journal archives

### PR DESCRIPTION
**Pull request by Steven-Eardley**
Use of ```Journal.all_in_doaj()``` might mean the withdraw script needs to be run more than once - the default page size is not enough to catch them all. (So I've doubled it in this case).

#323

### CODE REVIEW

File: portality/migrate/reapplication/delete_unreapplied.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
+from esprit import tasks, raw
+from portality import models
+
+'''Delete journals and corresponding articles with incomplete reapplications'''
+
+# Connection to the ES index
+conn = raw.make_connection(None, 'localhost', 9200, 'doaj')
```
Comment (emanuil-tolev) [inline portality/migrate/reapplication/delete_unreapplied.py]: You can guess the question here :). Why raw instead of through the DOAJ dao?

**Comment by emanuil-tolev** [comment]
> Use of Journal.all_in_doaj() might mean the withdraw script needs to be run more than once - the default page size is not enough to catch them all. (So I've doubled it in this case).

There are more than 10k journals in live.

Isn't the idea with paging that it will go through all of them, page by page? (So as to e.g. not load all of them in memory at once, or some other reason - but it should nonetheless iterate over all of them, even if you gave it a page size of 10.) Maybe I'm misunderstanding it here..

**Comment by Steven-Eardley** [comment]
I'm sure @richard-jones and I have run into this before - the behaviour seems to be that the page size does affect the number of results. I certainly had to run it twice when I tested it with a page size of 5000; first time getting just over 5000 results, the second getting about 2000 (with a total of 7k).

I've seen this in a few more places though, so if this outcome is a bug, it definitely needs looking into.

**Comment by emanuil-tolev** [comment]
Hm, right. This is just a script to be run once so a few more MB memory won't hurt, it's already got to load ~300 MB if I remember correctly how big the journal type is. Could you change the page_size to 20'000 then? Otherwise it will miss a few journals when we run it in live.

File: portality/migrate/reapplication/delete_unreapplied.py (hunk 1 above)
Comment (Steven-Eardley) [inline portality/migrate/reapplication/delete_unreapplied.py]: The archiving is handled through esprit, which I usually set up thusly - is there a better way to dump them all to file by query?

Comment (emanuil-tolev) [inline portality/migrate/reapplication/delete_unreapplied.py]: Ah you're using tasks.dump . No idea if there's a better way, but it seems reasonable, and presumably you've ran it successfully + we'll run it on staging eventually.

**Comment by emanuil-tolev** [comment]
:+1:

**Pull request merged**